### DISCUSSION TITLE
How to output function expressions for each layer of strata after modeling in GemPy?

### DISCUSSION THREAD

**Discussion by chiyue520** [gempy-project/gempy#826] [Q&A]
Hello! I have successfully modeled using GemPy, but now for some research purposes, I need to obtain the function expression for each layer of strata after modeling, that is, the surface fitting equation for GemPy modeling. What code command should I use?
Thanks a lot!!!

**Comment by AlexanderJuestel** [comment]
@chiyue520 that is a good question for @Leguark 

Also have a look at the GemPy Paper: https://gmd.copernicus.org/articles/12/1/2019/

**Reply by chiyue520** [reply]
Thanks for your suggestion! And please help out,thank you!!!@Leguark

**Comment by chiyue520** [comment]
@AlexanderJuestel Hello,based on your suggestion, I have read the paper in detail: https://gmd.copernicus.org/articles/12/1/2019/.
However, I did not find any relevant code to obtain the surface fitting function for each layer of the model. Therefore, I checked some cases on the Gempy official website(https://www.gempy.org/) and found that there is a function called "Compute at a given location", which does not seem to match my preset goals. I hope to obtain (1) the surface fitting function for each layer of strata, (2) or the x and y values at a given point to obtain the predicted z values of the model, then compare this value with the actual z value to predict model accuracy. What code command can achieve the above two requirements?
I really need your help AND thanks a lot!!!!!!

**Reply by AlexanderJuestel** [reply]
For (2), you may want to check out https://gemgis.readthedocs.io/en/latest/getting_started/tutorial/18_creating_depth_maps_from_gempy_models.html for obtaining the meshes of your model.

Is that what you need?
